README.adoc

```diff
@@ -1,3 +1,139 @@
+
+A link:http://docs.oracle.com/javase/1.5.0/docs/guide/javadoc/doclet/overview.html[Doclet] that allows exporting javadoc
+comments containing link:http://asciidoctor.org[AsciiDoc] text to AsciiDoc files, enabling combining the javadoc
+documentation into the AsciiDoc documentation for your Java project.
+
+Instead of exporting the javadoc comments to standard HTML files, the ExportDoclet exports them to AsciiDoc files,
+that in turn can converted to any output format supported by the link:http://asciidoctor.org[AsciiDoctor toolchain].
+
+== Building the ExportDoclet
+
+ExportDoclet is a Java Maven project that can be built directly from any IDE or using the following maven command:
+
+[source,bash]
+mvn clean package install
+
+That will build the tool and install it at your local maven repository, usually at the .m2 directory
+inside your home directory.
+
+== Using the ExportDoclet with javadoc by command line
+
+To manually call the javadoc tool by command line using the ExportDoclet you have to follow
+the next steps in a terminal:
+
+- Enter into the folder that contains the java files with javadoc comments you want to export
+- Execute `javadoc -docletpath $DOCLETJAR -doclet org.asciidoctor.ExportDoclet -d $OUTPUTDIR *.java`,
+where `$DOCLETJAR` is the path for the ExportDoclet jar file and `$OUTPUTDIR` is where you want
+to save the exported javadocs.
+
+== Using ExportDoclet with Maven javadoc plugin
+
+Using the ExportDoclet with Maven is pretty simple. You can include the maven-javadoc-plugin configuration inside the `<build>` section of project's pom.xml file, defining the required configuration for the javadoc command line tool that was used above. 
+
+The sub-sections below present some distinct configurations.
+
+=== Generating only the AsciiDoc files
+
+To generate just the AsciiDoc files and ignore the genration of the default HTML files, you can include the following code to the `<build>` section of your project's pom.xml.
+
+[source,xml]
+--
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-javadoc-plugin</artifactId>
+    <version>2.10.4</version>
+    <configuration>
+        <doclet>org.asciidoctor.ExportDoclet</doclet>
+        <docletArtifact>
+            <groupId>org.asciidoctor</groupId>
+            <artifactId>exportdoclet</artifactId>
+            <version>1.5.4-SNAPSHOT</version>
+        </docletArtifact>
+        <useStandardDocletOptions>true</useStandardDocletOptions>
+    </configuration>
+</plugin>
+--
+
+Now, to generate the javadocs in AsciiDoc files you can execute the `javadoc:javadoc` goal in Maven, using your IDE or the command line inside your project root directory:
+
+[source,bash]
+mvn javadoc:javadoc
```

Comment:
Sorry, but I've never used Gradle and I don't know how to change the example so that it will work correctly. About ant, there are some years that I don't use it and I really don't know how anymore.

I'm wondering if you could help in this point or we could leave it to the community to improve.